components/crypto/src/utils.rs

```diff
@@ -138,6 +122,124 @@ impl EncryptedKeys {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq)]
+pub struct Keys {
```

Comment:
Well, that's more like preferences question, but I think it would be better to have something like:

```rust
struct KeyPair {
  pub pk: PublicKey,
  pub sk: SecretKey,
}

pub struct Keys {
  pub consensus: KeyPair,
  pub service: KeyPair,
  pub identity: KeyPair,
}
```

Reply:
To argue in favor of `KeyPair`. `KeyPair` may encode the invariant that the secret key and public key correspond to each other, which is beneficial at least from the safety perspective. (Also, sodium's secret keys actually contain corresponding public keys inside them with offset 32; while the offset should probably be regarded as an implementation detail, ability to derive a public key from the secret one [is not](https://download.libsodium.org/doc/public-key_cryptography/public-key_signatures#extracting-the-seed-and-the-public-key-from-the-secret-key).)